agent/src/util/config_action.rs

```diff
@@ -341,6 +374,7 @@ mod config_action_tests {
                 stop_discovery_sender,
                 instance_map: instance_map.clone(),
                 finished_discovery_receiver,
+                last_generation: config.metadata.generation,
```

Comment:
Where is `last_generation` being used in this function? Does it need to be passed in?

Reply:
It's not currently being used. I only added it because the rust analyzer gives me the following if I don't add it. Alternatively, I'm sure there's probably a way we can make it optional in the struct?

```
missing structure fields:
- last_generation
```

Reply:
Ooops! The GitHub UI confused me and made me think this was part of a function definition. What you have is correct. Thanks

Reply:
👍🏻  I've done that way too many times 😀